core/ir/instr_shared.c

```diff
@@ -93,6 +93,10 @@ instr_create(void *drcontext)
     instr_set_isa_mode(instr, X64_CACHE_MODE_DC(dcontext) ? DR_ISA_AMD64 : DR_ISA_IA32);
 #elif defined(ARM)
     instr_set_isa_mode(instr, dr_get_isa_mode(dcontext));
+#elif defined(AARCH64)
+    instr_set_isa_mode(instr, DR_ISA_ARM_A64);
+#elif defined(RISCV64)
+    instr_set_isa_mode(instr, DR_ISA_RV64IMAFDC);
```

Comment:
Let's have `#else #error` to be sure this gets updated on future arches.

Although: the ARM version of getting it from the dcontext seems like it should work for all arches: so `s/if defined(ARM)/se/` instead is better, keeping the default arch ISA in just one place.

Reply:
Ok, ARM and all future arches set instr_t isa_mode from dcontext now.
I noticed we don't check the return value of instr_set_isa_mode().
Should we? Or is there a reason we don't?

Reply:
I meant that a64 and riscv would not have separate code either: only x86 (maybe remove x86 too if you convince yourself it's the same also: but it looks complicated).

Reply:
Checking return value sounds good.

Reply:
Ah, I had it right before.
Putting all arches except x86 in the #else then.
I'll have a look if x86 can be included as well, and add checks for instr_set_isa_mode() returned value.

Reply:
Had to keep the x86-related #ifdef.
Added a check for the return value of instr_set_isa_mode() using CLIENT_ASSERT(), following how other functions do it (out of curiosity: why is it "CLIENT" ASSERT? and not just "ASSERT"?).